osu.Game/Screens/Menu/IntroScreen.cs

```diff
@@ -201,7 +209,14 @@ public override void OnResuming(IScreen last)
             // we also handle the exit transition.
             if (MenuVoice.Value)
             {
-                seeya.Play();
+                // ensure samples have been updated after resume before playing.
```

Comment:
What's the deal with this? Does playing a `SkinnableSound` before it has been updated once not work or something?

Reply:
It will play the last sample retrieved before the scheduled `skinChange` executes in `PoolableSkinnableSample`.

Since, once a screen gets resumed, `OnResume` executes before the screen itself becomes alive and and update its hierarchy next frame, therefore we need to `ScheduleAfterChildren`.

Reply:
I think you need to rewrite the inline comment to read as such. Specifically mention the reload flow on skin change as that is important and not really conveyed in your comment.

Reply:
Has been rewritten and rescoped:
![CleanShot 2022-03-10 at 17 27 56](https://user-images.githubusercontent.com/22781491/157682583-edf70955-1aaf-4fca-8289-55dcce238fd3.png)